src/Components/Web/src/Virtualization/InfiniteScroll.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Virtualization
+{
+    /// <summary>
+    /// Provides functionality for rendering a virtualized list from a source of unknown size.
+    /// </summary>
+    /// <typeparam name="TItem"></typeparam>
+    public sealed class InfiniteScroll<TItem> : ComponentBase
+    {
+        /// <summary>
+        /// Gets or sets the item template for the list.
+        /// </summary>
+        [Parameter]
+        public RenderFragment<TItem> Item { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the template for the footer of the list.
+        /// </summary>
+        [Parameter]
+        public RenderFragment<int> Footer { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the function retrieving items given a start index.
+        /// </summary>
+        [Parameter]
+        public ItemsProviderDelegate<TItem> ItemsProvider { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the item count requested to the <see cref="ItemsProvider"/>.
+        /// </summary>
+        [Parameter]
+        public int RequestedItemCount { get; set; } = 10;
+
+        /// <summary>
+        /// Gets the size of each item in pixels.
+        /// </summary>
+        [Parameter]
+        public float ItemSize { get; set; }
```

Comment:
Previously we talked about InfiniteScroll being able to work without knowing the item size, because it's just going to stack them up and doesn't support random access into the middle of the list.

Do you think that's viable?